### PR TITLE
DOCS-1318: Document `DoCommand` for Power Sensor

### PR DESCRIPTION
**Pull request by sguequierre**
* Adds DoCommand for Power Sensor

### CODE REVIEW

File: docs/components/power-sensor/_index.md [hunk 1]
```diff
@@ -297,6 +297,59 @@ readings, err := myPowerSensor.Readings(context.Background(), nil)
...
+
+```python {class="line-numbers linkable-line-numbers"}
+my_sensor = PowerSensor.from_robot(robot=robot, name="my_power_sensor")
+
+reset_dict = {
+  "command": "reset",
+  "example_param": 30
+}
+
+do_response = await my_sensor.do_command(reset_dict)
```
Comment (andf-viam) [inline docs/components/power-sensor/_index.md:324]:
```suggestion
my_power_sensor = PowerSensor.from_robot(robot=robot, name="my_power_sensor")

reset_dict = {
  "command": "reset",
  "example_param": 30
}

do_response = await my_power_sensor.do_command(reset_dict)
```

To match other examples on page.

Comment (andf-viam) [reply docs/components/power-sensor/_index.md:324]: Ah, let's update lines 268 and 271 above with same: `my_power_sensor`

File: docs/components/power-sensor/_index.md [hunk 2]
```diff
@@ -297,6 +297,59 @@ readings, err := myPowerSensor.Readings(context.Background(), nil)
...
+
+**Returns:**
+
+- [(map[string]interface{})](https://go.dev/blog/maps): Result of the executed command.
+- [(error)](https://pkg.go.dev/builtin#error): An error, if one occurred.
+
+```go {class="line-numbers linkable-line-numbers"}
+mySensor, err := powersensor.FromRobot(robot, "my_power_sensor")
+
+resp, err := mySensor.DoCommand(ctx, map[string]interface{}{"command": "reset", "example_param": 30})
```
Comment (andf-viam) [inline docs/components/power-sensor/_index.md:345]:
```suggestion
myPowerSensor, err := powersensor.FromRobot(robot, "my_power_sensor")

resp, err := myPowerSensor.DoCommand(ctx, map[string]interface{}{"command": "reset", "example_param": 30})
```

To match other examples on page.

File: docs/components/power-sensor/_index.md [hunk 3]
```diff
@@ -297,6 +297,59 @@ readings, err := myPowerSensor.Readings(context.Background(), nil)
...
+
+For more information, see the [Python SDK Docs](https://python.viam.dev/autoapi/viam/components/power_sensor/client/index.html#viam.components.power_sensor.client.PowerSensorClient.do_command).
+
+{{% /tab %}}
+{{% tab name="Go" %}}
+
+**Parameters:**
+
+- `ctx` [(Context)](https://pkg.go.dev/context): A Context carries a deadline, a cancellation signal, and other values across API boundaries.
+- `cmd` [(map[string]interface{})](https://go.dev/blog/maps): The command to execute.
```
Comment (andf-viam) [inline docs/components/power-sensor/_index.md:335]: Question: I'm vbad at go, but should this be `cmd` --> `command` to match usage in the Go example below? If so, I see this elsewhere in the docs (as `cmd`) ([SLAM](https://docs.viam.com/services/slam/#docommand) for example) as well.

**Review by andf-viam — APPROVED** [review]
LGTM, just some nits and a question. Thanks!

File: docs/components/power-sensor/_index.md [hunk 4]
```diff
@@ -297,6 +297,59 @@ readings, err := myPowerSensor.Readings(context.Background(), nil)
...
+- [(map[string]interface{})](https://go.dev/blog/maps): Result of the executed command.
+- [(error)](https://pkg.go.dev/builtin#error): An error, if one occurred.
+
+```go {class="line-numbers linkable-line-numbers"}
+mySensor, err := powersensor.FromRobot(robot, "my_power_sensor")
+
+resp, err := mySensor.DoCommand(ctx, map[string]interface{}{"command": "reset", "example_param": 30})
+```
+
+For more information, see the [Go SDK Code](https://github.com/viamrobotics/rdk/blob/main/resource/resource.go).
```
Comment (sguequierre) [inline docs/components/power-sensor/_index.md:348]:
```suggestion
For more information, see the [Go SDK Code](https://pkg.go.dev/go.viam.com/rdk@v0.12.0/resource#Resource).
```

File: docs/components/power-sensor/_index.md (hunk 3 above)
Comment (sguequierre) [reply docs/components/power-sensor/_index.md:335]: It is named cmd, but the example demonstrates feeding it a mapping with an example command named command! See https://pkg.go.dev/go.viam.com/rdk@v0.12.0/resource#Resource (which should really be linked to here, updating). Is that still confusing?

File: docs/components/power-sensor/_index.md [hunk 5]
```diff
@@ -265,10 +265,10 @@ If a sensor is not configured to have a measurement or fails to read a piece of
 For more information, see the [Python SDK Docs](https://python.viam.dev/autoapi/viam/components/power_sensor/client/index.html#viam.components.power_sensor.client.PowerSensorClient.get_readings).
 
 ```python
-my_sensor = PowerSensor.from_robot(robot=robot, name='my_power_sensor')
+my_power_sensor = PowerSensor.from_robot(robot=robot, name='my_power_sensor')
 
 # Get the readings provided by the sensor.
-readings = await my_sensor.get_readings()
+readings = await my_power_sensor.get_readings()
```
Comment (sguequierre) [inline docs/components/power-sensor/_index.md:271]: @andf-viam Updated!

File: docs/components/power-sensor/_index.md (hunk 3 above)
Comment (andf-viam) [reply docs/components/power-sensor/_index.md:335]: No that makes sense! I think the updated link helps clarify also. Thanks much!!

File: docs/components/power-sensor/_index.md (hunk 5 above)
Comment (andf-viam) [reply docs/components/power-sensor/_index.md:271]: LGTM!

**Comment by JessamyT** [comment]
Please add DoCommand to the API table includes!

**Comment by npentrel** [comment]
Updating this branch to test something. since it seems ready to merge I hope that won't mess up anything :)

**Pull request merged**